### DISCUSSION TITLE
How to set Termination Protection for an EC2 Instance?

### DISCUSSION THREAD

**Discussion by mikelhamer** [aws/aws-cdk#21779] [Q&A] [answered]
I'm not seeing any way in the EC2 construct to set Termination Protection. Is this possible?

To clarify, I'm talking about the setting for the actual EC2 Instance, not the Stack.

**Comment by PatMyron** [comment]
Present in [L1](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.CfnInstance.html#disableapitermination) but not [L2](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Instance.html) construct, so set via [escape hatch](https://docs.aws.amazon.com/cdk/v2/guide/cfn_layer.html#develop-customize-escape)

**Comment by khushail** [comment, marked as answer]
Adding more reference to @PatMyron answer, the property is -

```
disableApiTermination

Type: boolean | [IResolvable](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.IResolvable.html) (optional)

If you set this parameter to true , you can't terminate the instance using the Amazon EC2 console, CLI, or API;

otherwise, you can. To change this attribute after launch, use [ModifyInstanceAttribute](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_ModifyInstanceAttribute.html) . Alternatively, if you set InstanceInitiatedShutdownBehavior to terminate , you can terminate the instance by running the shutdown command from the instance.

Default: false
```

like this -
```ts
    const instance = new ec2.CfnInstance(this, "myInstance", {
          disableApiTermination :true,
    })
```
